benchmarks/prescribe_dike_injection/prescribed_dike_injection_heat.cc

```diff
@@ -0,0 +1,137 @@
+        (this->get_parameters().enable_prescribed_dilation)
+        ? material_model_outputs.template get_additional_output<MaterialModel::PrescribedPlasticDilation<dim> >()
+        : nullptr;
+
+      // Add the latent heat source term corresponding to prescribed injection
+      // terms in Stokes equations to the rhs of energy conservation equation.
+      double dike_injection_fraction = 0.0;
+
+      for (unsigned int q=0; q<heating_model_outputs.heating_source_terms.size(); ++q)
+        {
+          heating_model_outputs.heating_source_terms[q] = 0.0;
+          heating_model_outputs.lhs_latent_heat_terms[q] = 0.0;
+          heating_model_outputs.rates_of_temperature_change[q] = 0.0;
+          if (prescribed_dilation != nullptr)
+            {
+              // User-defined or timestep-dependent injection ratio
+              if (this->simulator_is_past_initialization())
+                dike_injection_fraction = prescribed_dilation->dilation[q] * this->get_timestep();
+
+              if (dike_material_injection_fraction != 0.0)
+                dike_injection_fraction = dike_material_injection_fraction;
+
+              // adding the laten heat source team
+              heating_model_outputs.heating_source_terms[q] = dike_injection_fraction * 
+                                                              prescribed_dilation->dilation[q] * 
+                                                              (latent_heat_of_crystallization + 
+                                                              (temperature_of_injected_material - 
+                                                              material_model_inputs.temperature[q]) * 
+                                                              material_model_outputs.densities[q] * 
+                                                              material_model_outputs.specific_heat[q]);
+            }
+        }
+    }
+
+    template <int dim>
+    void
+    LatentHeatDikeInjection<dim>::declare_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Heating model");
+      {
+        prm.enter_subsection("Latent heat dike injection");
+        {
+          prm.declare_entry ("Latent heat of crystallization", "1.1e9",
+                             Patterns::Double(0),
+                             "The latent heat of crystallization that is released when material "
+                             "is injected into the model. "
+                             "Units: J/m$^3$.");
+          prm.declare_entry ("Temperature of the injected material", "1273",
+                             Patterns::Double(0),
+                             "The temperature of the material injected into the model. "
+                             "Units: K.");
+          prm.declare_entry("Dike material injection fraction", "0.0",
+                            Patterns::Double(0),
+                            "Amount of new injected material from the dike. Units: none.");
```

Comment:
Can you read this from the prescribed_dike_injection material model section?

Reply:
I don't find an easy way to do it. Any suggestions?